modules/common/src/main/scala/com.snowplowanalytics.snowplow.enrich/common/enrichments/EnrichmentManager.scala

```diff
@@ -92,8 +88,53 @@ object EnrichmentManager {
                enriched.pii = pii.asString
              }
            }
+      _ <- validateEnriched(enriched, raw, processor, client)
     } yield enriched
 
+  private val atomicSchema: SchemaKey = SchemaKey("com.snowplowanalytics.snowplow", "atomic", "jsonschema", SchemaVer.Full(1, 0, 0))
+
+  private def validateEnriched[F[_]](
+    enriched: EnrichedEvent,
+    raw: RawEvent,
+    processor: Processor,
+    client: Client[F, Json]
+  )(
+    implicit
+    M: Monad[F],
+    L: RegistryLookup[F],
+    C: Clock[F]
+  ): EitherT[F, BadRow, Unit] =
+      enriched.toShreddedEvent
+        .leftMap(err =>
+          EnrichmentManager.buildEnrichmentFailuresBadRow(
+            NonEmptyList.one(
+              FailureDetails.EnrichmentFailure(Some(EnrichmentInformation(atomicSchema, identifier = "shredded_validator")),
+                                               EnrichmentFailureMessage.Simple(err.getMessage)
+              )
+            ),
+            toPartiallyEnrichedEvent(enriched),
+            toRawEvent(raw),
+            processor
+          )
+        )
+        .toEitherT[F]
+        .flatMap(shreded =>
+          client
+            .check(
+              SelfDescribingData(atomicSchema, shreded)
+            )
+            .leftMap(err =>
+              buildSchemaViolationsBadRow(
```

Comment:
In my opinion this should not be a schema violation. Schema violations are for the input SDJs that we receive from trackers. If we are here it means that the enrichments added/mutated something invalid, so this should be an enrichment failure.